lang/ast/structs.go

```diff
@@ -8318,23 +8222,10 @@ func (obj *ExprCall) Func() (interfaces.Func, error) {
 	}, nil
 }
 
-// SetValue here is used to store the result of the last computation of this
-// expression node after it has received all the required input values. This
-// value is cached and can be retrieved by calling Value.
-func (obj *ExprCall) SetValue(value types.Value) error {
-	if err := obj.typ.Cmp(value.Type()); err != nil {
-		return err
-	}
-	obj.V = value
-	return nil
-}
-
 // Value returns the value of this expression in our type system. This will
 // usually only be valid once the engine has run and values have been produced.
 // This might get called speculatively (early) during unification to learn more.
 // It is often unlikely that this kind of speculative execution finds something.
-// This particular implementation of the function returns the previously stored
-// and cached value as received by SetValue.
 func (obj *ExprCall) Value() (types.Value, error) {
 	if obj.V == nil {
 		return nil, fmt.Errorf("func value does not yet exist")
```

Comment:
Why fail when the value is `nil`? As the comment says, `Value()` is sometimes called speculatively during unification, so it should be fine to return `nil` in that case.

The test `TestAstFunc2/test_#0_` currently fails in CI with this error message.